modules/solace/src/test/java/org/testcontainers/containers/SolaceContainerSMFTest.java

```diff
@@ -0,0 +1,169 @@
+package org.testcontainers.containers;
+
+import com.solacesystems.jcsmp.BytesXMLMessage;
+import com.solacesystems.jcsmp.JCSMPException;
+import com.solacesystems.jcsmp.JCSMPFactory;
+import com.solacesystems.jcsmp.JCSMPProperties;
+import com.solacesystems.jcsmp.JCSMPSession;
+import com.solacesystems.jcsmp.JCSMPStreamingPublishCorrelatingEventHandler;
+import com.solacesystems.jcsmp.TextMessage;
+import com.solacesystems.jcsmp.Topic;
+import com.solacesystems.jcsmp.XMLMessageConsumer;
+import com.solacesystems.jcsmp.XMLMessageListener;
+import com.solacesystems.jcsmp.XMLMessageProducer;
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.utility.MountableFile;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Tomasz Forys
+ */
+public class SolaceContainerSMFTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolaceContainerSMFTest.class);
+
+    private static final Topic TOPIC = JCSMPFactory
+        .onlyInstance()
+        .createTopic(SolaceContainerTestProperties.TOPIC_NAME);
+
+    @Test
+    public void testSolaceContainerWithSimpleAuthentication() {
+        try (
+            // solaceContainerSetup {
+            SolaceContainer solace = new SolaceContainer(
+                SolaceContainerTestProperties.getImageName().asCanonicalNameString()
+            )
+                .withCredentials("user", "pass")
+                .withTopic(SolaceContainerTestProperties.TOPIC_NAME, Service.SMF)
+                .withVpn("test_vpn")
+            // }
+        ) {
+            solace.start();
+            JCSMPSession session = createSessionWithBasicAuth(solace);
+            Assertions.assertThat(session).isNotNull();
+            Assertions.assertThat(consumeMessageFromSolace(session)).isEqualTo(SolaceContainerTestProperties.MESSAGE);
+            session.closeSession();
+        }
+    }
+
+    @Test
+    public void testSolaceContainerWithCertificates() {
+        try (
+            // solaceContainerUsageSSL {
+            SolaceContainer solace = new SolaceContainer(SolaceContainerTestProperties.getImageName())
```

Comment:
This is being displayed in the docs. In order to make it clear, we should use the raw image name "solace/solace-pubsub-standard:10.2"